README.md

```diff
@@ -202,6 +204,20 @@ func Down(tx *sql.Tx) error {
 }
 ```
 
+## Go Migrations from a precompiled plugins
+
+1. Import `github.com/pressly/goose`
+2. Load migration plugins with LoadMigrationPlugins
+4. Run goose command, ie. `goose.Up(db *sql.DB, dir string)`
+
+A migration plugin can be created from a go template generated by goose.
+Golang plugin system requires that the plugin's namespace is main.
+To compile a plugin one can use the following command:
+
+```bash
+go build -buildmode=plugin -o 002_next.so 002_next.sql
```

Comment:
This is all great, I really like it!

We need to explain really well that you can either compile migration functions into your own binary -- or use plugins as an alternative.